### PR TITLE
Add object type

### PR DESCRIPTION
**Pull request by kangkaisen**
For https://github.com/apache/incubator-doris/pull/1610

Add a new type: Object. Currently, it's mainly for complex aggregate metrics(HLL , Bitmap).

The Object type has the following  constraints：
1 Object type could not as key column type
2 Object type doesn't support all indices (BloomFilter, short key, zone map, invert index)
3 Object type doesn't support filter and group by

In the implementation：

The Object type reuse the `StringValue` and `StringVal`, because in storage engine, the Object type  is binary, it has a pointer and length.

### CODE REVIEW

File: gensrc/proto/segment_v2.proto [hunk 1]
```diff
@@ -53,6 +53,7 @@ enum EncodingTypePB {
     RLE = 4;
     DICT_ENCODING = 5;
     BIT_SHUFFLE = 6;
+    BINARY_PLAIN_ENCODING = 7;
```
[the remark this answers is not in the document]
Comment (kangkaisen) [reply gensrc/proto/segment_v2.proto]: done

**Comment by kangkaisen** [comment]
Update.
Use bitmap type in FE and convert to object type in Thrift.

File: fe/src/main/jflex/sql_scanner.flex [hunk 2]
```diff
@@ -218,6 +218,7 @@ import org.apache.doris.common.util.SqlUtils;
         keywordMap.put("not", new Integer(SqlParserSymbols.KW_NOT));
         keywordMap.put("null", new Integer(SqlParserSymbols.KW_NULL));
         keywordMap.put("nulls", new Integer(SqlParserSymbols.KW_NULLS));
+        keywordMap.put("object", new Integer(SqlParserSymbols.KW_OBJECT));
```
[the remark this answers is not in the document]
Comment (kangkaisen) [reply fe/src/main/jflex/sql_scanner.flex]: >Do we need to expose OBJECT to Doris USER? I think we should expose specified type to user, such as HLL, GEO_SHAPE, BITMAP?

OK

**Review by imay — COMMENTED** [review]
Should update correspond document.

**Comment by kangkaisen** [comment]
1 Object type support v1
2 Change the bitmap_union type to bitmap
3 Add check for bitmap_count function
4 Update the doc

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**